run.slurm

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#SBATCH --job-name=instructpix2pix-sdxl
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1          # crucial - only 1 task per dist per node!
+#SBATCH --cpus-per-task=96
+#SBATCH --gres=gpu:8
+#SBATCH --exclusive
+#SBATCH --partition=production-cluster
+#SBATCH --output=/admin/home/suraj/logs/maskgit-imagenet/%x-%j.out
+
+set -x -e
+
+source /admin/home/suraj/.bashrc
+source /fsx/suraj/miniconda3/etc/profile.d/conda.sh
+conda activate muse
+
+echo "START TIME: $(date)"
+
+REPO=/admin/home/suraj/code/instructpix2pix-sdxl
+OUTPUT_DIR=/fsx/suraj/instructpix2pix-sdxl
+LOG_PATH=$OUTPUT_DIR/main_log.txt
+ACCELERATE_CONFIG_FILE="$OUTPUT_DIR/${SLURM_JOB_ID}_accelerate_config.yaml.autogenerated"
+
+mkdir -p $OUTPUT_DIR
+touch $LOG_PATH
+pushd $REPO
+
+
+GPUS_PER_NODE=8
+NNODES=$SLURM_NNODES
+NUM_GPUS=$((GPUS_PER_NODE*SLURM_NNODES))
+
+# so processes know who to talk to
+MASTER_ADDR=$(scontrol show hostnames $SLURM_JOB_NODELIST | head -n 1)
+MASTER_PORT=6000
+
+
+# Auto-generate the accelerate config
+cat << EOT > $ACCELERATE_CONFIG_FILE
+compute_environment: LOCAL_MACHINE
+deepspeed_config: {}
+distributed_type: MULTI_GPU
+fsdp_config: {}
+machine_rank: 0
+main_process_ip: $MASTER_ADDR
+main_process_port: $MASTER_PORT
+main_training_function: main
+num_machines: $SLURM_NNODES
+num_processes: $NUM_GPUS
+use_cpu: false
+EOT
+
+
+export MODEL_ID="stabilityai/stable-diffusion-xl-base-1.0"
+
+PROGRAM="train_instruct_pix2pix_sdxl.py \
+    --pretrained_model_name_or_path=$MODEL_ID \
+    --pretrained_vae_model_name_or_path=madebyollin/sdxl-vae-fp16-fix \
+    --dataset_path='pipe:aws s3 cp s3://muse-datasets/instructpix2pix-clip-filtered-wds/{000000..000062}.tar -' \
+    --use_ema \
+    --enable_xformers_memory_efficient_attention \
+    --resolution=256 --random_flip \
+    --per_gpu_batch_size=16 --gradient_accumulation_steps=4 \
+    --num_workers=4 \
```

Comment:
num_workers=48 is too much. That's 48 per GPU, so 384 in total. With those many workers, we'll send a lot of simultaneous requests to `s3`, which will block it, which results in broken pipe error.